[DDP Comm Hook] Add debugging communication hooks to ddp_comm_hooks.rst

wayi1 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #64352
* #64344

as title

Differential Revision: [D30694089](https://our.internmc.facebook.com/intern/diff/D30694089/)

Commit:
[DDP Comm Hook] Add debugging communication hooks to ddp_comm_hooks.rst

as title

Differential Revision: [D30694089](https://our.internmc.facebook.com/intern/diff/D30694089/)

[ghstack-poisoned]

Commit:
Update on "[DDP Comm Hook] Add debugging communication hooks to ddp_comm_hooks.rst"


as title

Differential Revision: [D30694089](https://our.internmc.facebook.com/intern/diff/D30694089/)

[ghstack-poisoned]

rohan-varma commented on docs/source/ddp_comm_hooks.rst:
```diff
@@ -82,6 +82,18 @@ PowerSGD Hooks
+.. currentmodule:: torch.distributed.algorithms.ddp_comm_hooks.debugging_hooks
+
+.. warning ::
+    Debugging communication hooks do not necessarily output the correct results.
```
Maybe expand on this a bit, and say that debugging comm. hooks are mostly meant for perf optimization tests, and thus shouldn't be used for accuracy purposes?

SciPioneer replied:
I think the purpose has been emphasized in both the first sentence of this section as well as the docstring.

Commit:
Update on "[DDP Comm Hook] Add debugging communication hooks to ddp_comm_hooks.rst"


as title

Differential Revision: [D30694089](https://our.internmc.facebook.com/intern/diff/D30694089/)

[ghstack-poisoned]

Commit:
Update on "[DDP Comm Hook] Add debugging communication hooks to ddp_comm_hooks.rst"


as title

Differential Revision: [D30694089](https://our.internmc.facebook.com/intern/diff/D30694089/)

[ghstack-poisoned]
